pkg/cedar/core/budget.go

```diff
@@ -21,17 +21,16 @@ func (c *Client) GetBudgetBySystem(ctx context.Context, cedarSystemID string) ([
 		return nil, cedarcoremock.NoSystemFoundError()
 	}
 	cedarSystem, err := c.GetSystem(ctx, cedarSystemID)
+	if err != nil {
```

Comment:
Was this causing nil dereference errors when trying to fetch the `VersionID` and/or `VersionID.Ptr()`?

Reply:
i don't believe we were receiving any errors, but it can happen if we check the error after those lines (guessing this was mainly just a typo on err check placement)